core/registry.js

```diff
@@ -34,81 +47,80 @@ goog.requireType('Blockly.ToolboxItem');
  *
  * @type {Object<string, Object<string, function(new:?)>>}
  */
-Blockly.registry.typeMap_ = Object.create(null);
+const typeMap = Object.create(null);
+/** @private */
+exports.typeMap_ = typeMap;
 
 /**
  * The string used to register the default class for a type of plugin.
  * @type {string}
  */
-Blockly.registry.DEFAULT = 'default';
+const DEFAULT = 'default';
+exports.DEFAULT = DEFAULT;
 
 /**
  * A name with the type of the element stored in the generic.
  * @param {string} name The name of the registry type.
  * @constructor
  * @template T
  */
-Blockly.registry.Type = function(name) {
+const Type = function(name) {
   /**
    * @type {string}
    * @private
    */
   this.name_ = name;
 };
+exports.Type = Type;
```

Comment:
This seems like it may be one of the cases where Type ought to be a class in its own file and moved out of this one, and we should add it to the list of weird files to follow up on. Because this isn't just a static function on the registry namespace like many of the other files have.

Reply:
Added a note about this to the tracking issue, I agree it seems reasonable to split it out into its own file.